usr/share/rear/conf/Arch/i386.conf

```diff
@@ -0,0 +1 @@
+COPY_AS_IS=( "${COPY_AS_IS[@]}" '/etc/ca-certificates/*' )
```

Comment:
I would use `/etc/ca-certificates*` (without quotes and slash) to cover the case where this directory doesn't exist (as we use nullglob).

Also, maybe it is a good idea for other distros as well and worth putting into a more generic file than just for Arch Linux.

Reply:
I've took [Linux.conf](https://github.com/rear/rear/blob/master/usr/share/rear/conf/GNU/Linux.conf#L249) as a reference, and copied syntax from there, so maybe quotes should be removed there as well ?

Currently I have booted 3 distros Debian, SLES12-SP2, and Arch. CA certificate directory is different on each of them :-).

Thinking of it, maybe _/etc/ca-certificates/*_ should be put to [Linux.conf](https://github.com/rear/rear/blob/master/usr/share/rear/conf/GNU/Linux.conf)...

Reply:
Yes, I also realized that we have both versions, with and without quotes. I think without is better because then the pattern is evaluated immediately. Also, if the directory doesn't exist then it *disappears* completely. Therefore it is safe to list all the the different possible directories and only the ones that actually exist will be listed in the array.

For this trick to work you have to specify the pattern without a trailing slash or without a trailing `/*`, for example like this:

```bash
/etc/ca-[c]ertificates
```

That will match a file or directory named `/etc/ca-certificates`, but only if it exists 😄 

You'll find plenty more good (and bad) examples if you do a `git grep "COPY_AS_IS.*\*"`

Reply:
@gozora I'm good with the extra line(s). Perhaps change the code slightly like @schlomo suggested.

Reply:
Everybody OK if I remove Arch/i386.conf and edit [Linux.conf](https://github.com/rear/rear/blob/master/usr/share/rear/conf/GNU/Linux.conf) as follows instead ?
```
-   COPY_AS_IS=( "${COPY_AS_IS[@]}" '/etc/ssl/certs/*' '/etc/pki/*' '/usr/lib/ssl/*' '/usr/share/ca-certificates/*'  )
+   COPY_AS_IS=( "${COPY_AS_IS[@]}" /etc/ssl/certs /etc/pki /usr/lib/ssl /usr/share/ca-certificates  /etc/ca-certificates )
```

V.

Reply:
@schlomo something like this:

```
-   COPY_AS_IS=( "${COPY_AS_IS[@]}" '/etc/ssl/certs/*' '/etc/pki/*' '/usr/lib/ssl/*' '/usr/share/ca-certificates/*'  )
+   COPY_AS_IS=( "${COPY_AS_IS[@]}" /etc/ssl/[c]erts /etc/[p]ki /usr/lib/[s]sl /usr/share/[c]a-certificates  /etc/[c]a-certificates )
```
?

Sorry for such questions, but this is completely new thing to me ...

V.

Reply:
Yes, like this. This is some of the more useful Bash magic tricks:
![image](https://user-images.githubusercontent.com/101384/48766696-50955600-ecb5-11e8-95c4-a5a944b23e52.png)

Reply:
Having `nullglob` in ReaR by default is a double-edged sword,
'grep' for `nullglob` in the comments of the code to get some places
where I had to fix things because of its unexpected behaviour
compared to what one gets on the command line...

Reply:
Yes, it requires one to think about the potential results of a pattern expansion. I think that this is a good side-effect as it forces us to write more robust code. I just now looked at all the `nullglob` references and most seem to me like a "note to self" comment.

IMHO
```bash
MEMTEST_BIN=$(ls -d /boot/memtest86+-* 2>/dev/null | tail -1)
if [[ "$MEMTEST_BIN" != "." && -r "$MEMTEST_BIN" ]]; then
```
is more interesting as an example. I think that the following code would be cleaner and easier to understand:
```bash
MEMTEST_BIN=( $( ls -rd /boot/memtest86+-* ) ) # get highest memtest86+ version
if [[ -f "$MEMTEST_BIN" ]]; then
```
That way one doesn't fight against `nullglob` but utilizes it to make the code more simple.

IMHO it will cover all eventualities (assuming that as root we can always read the file content).

Reply:
I recommend to use e.g.
<pre>
git log -p --follow usr/share/rear/output/USB/Linux-i386/300_create_extlinux.sh
</pre>
to see when and by whom what code was actually made ;-)
As often the actual code is very old - also in this case
(and I am unable to do all the needed forensics to fully understand
what each piece of code is actually meant to do to properly fix all).

I think this example plus "this is completely new thing to me" in
https://github.com/rear/rear/pull/1971#discussion_r234928422
shows that `nullglob` in ReaR by default can result problematic code,
in particular for our users who may adapt and enhance ReaR scripts
as they need it in their particular cases but who are not aware of
our non-standard bash behaviour, e.g. as in
<pre>
# ( SOMEFILES="$( ls /etc/maynotexist/* )" ; echo "SOMEFILES='$SOMEFILES'" )
ls: cannot access '/etc/maynotexist/*': No such file or directory
SOMEFILES=''

# ( shopt -s nullglob ; SOMEFILES="$( ls /etc/maynotexist/* )" ; echo "SOMEFILES='$SOMEFILES'" )
SOMEFILES='.editorconfig
.git
.github
.gitignore
.travis.yml
COPYING
MAINTAINERS
Makefile
README.adoc
doc
etc
packaging
tests
usr
var'
</pre>
where the first case works fail safe but not the second one.

Reply:
@jsmeix are you asking about passing the `shopt` settings into subshells? It seems to "just work":
![image](https://user-images.githubusercontent.com/101384/48782620-e1345c00-ecde-11e8-8482-aec3b11982f8.png)

https://stackoverflow.com/questions/24173875/is-there-a-way-to-export-bash-shell-options-to-subshell suggests to `export BASHOPTS` which we should add if somebody has a problem in this area.

ReaR uses *enhanced Bash* and everybody coding for ReaR will have to learn that. We maintainers are of course happy to help with that.

Reply:
I used the subshell only to not change the behaviour
of my "outer" shell which is my usual workday shell.